android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/activity/ActivitySchemaCache.java

```diff
@@ -0,0 +1,156 @@
+package org.sagebionetworks.bridge.android.manager.activity;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.common.collect.Maps;
+
+import org.sagebionetworks.bridge.rest.model.Activity;
+import org.sagebionetworks.bridge.rest.model.ActivityType;
+import org.sagebionetworks.bridge.rest.model.CompoundActivity;
+import org.sagebionetworks.bridge.rest.model.SchemaReference;
+import org.sagebionetworks.bridge.rest.model.SurveyReference;
+import org.sagebionetworks.bridge.rest.model.TaskReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * A cache of mappings from activities' identifiers (tasks, surveys, compound activities) to
+ * (SchemaReference, SurveyReference). This is to help identify how an activity's return data should
+ * be formatted.
+ */
+public class ActivitySchemaCache {
+    private static final Logger LOG = LoggerFactory.getLogger(ActivitySchemaCache.class);
+    @NonNull
+    private final Map<String, SchemaReference> taskIdentifierToSchema;
+    @NonNull
+    private final Map<String, SurveyReference> surveyIdentifierToSurvey;
+    @NonNull
+    private final Map<String, String> surveyGuidToIdentifier;
+    @NonNull
+    private final Map<String, CompoundActivity> taskIdentifierToCompoundActivity;
+    @NonNull
+    private final Map<String, ActivityType> identifierToActivityType;
+
+
+    public ActivitySchemaCache() {
+        taskIdentifierToSchema = Maps.newHashMap();
+        surveyIdentifierToSurvey = Maps.newHashMap();
+        surveyGuidToIdentifier = Maps.newHashMap();
+        taskIdentifierToCompoundActivity = Maps.newHashMap();
+        identifierToActivityType = Maps.newHashMap();
+
+        addTasks();
+    }
+
+    private void addTasks() {
```

Comment:
Before I can approve this PR, let's add methods to BridgeDataProvider to set up the ActivitySchemaCache.  I think that would be the easiest time to set it up, when we create it in mPower.  Of course, we could always use a JSON file, but I think it will make more sense in code.